aws/resource_aws_ram_resource_share_accepter.go

```diff
@@ -0,0 +1,298 @@
+	}
+
+	log.Println("[DEBUG] Accept RAM resource share invitation request:", in)
+	out, err := conn.AcceptResourceShareInvitation(in)
+	if err != nil {
+		return fmt.Errorf("Error accepting RAM resource share invitation: %s", err)
+	}
+
+	d.SetId(*out.ResourceShareInvitation.ResourceShareInvitationArn)
+
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{ram.ResourceShareInvitationStatusPending},
+		Target:  []string{ram.ResourceShareInvitationStatusAccepted},
+		Refresh: resourceAwsRamResourceShareAccepterStateRefreshFunc(
+			conn, *out.ResourceShareInvitation.ResourceShareInvitationArn),
+		Timeout: d.Timeout(schema.TimeoutCreate),
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for RAM resource share (%s) state: %s", d.Id(), err)
+	}
+
+	return resourceAwsRamResourceShareAccepterRead(d, meta)
+}
+
+func resourceAwsRamResourceShareGetInvitationARN(conn *ram.RAM, resourceShareARN string) (*string, error) {
+	var nextToken string
+	for {
+		input := &ram.GetResourceShareInvitationsInput{
+			MaxResults:        aws.Int64(int64(500)),
+			ResourceShareArns: aws.StringSlice([]string{resourceShareARN}),
+		}
+		if nextToken != "" {
+			input.NextToken = aws.String(nextToken)
+		}
+		out, err := conn.GetResourceShareInvitations(input)
+		if err != nil {
+			return nil, err
+		}
+		for _, invitation := range out.ResourceShareInvitations {
+			if *invitation.Status == ram.ResourceShareInvitationStatusPending {
+				return invitation.ResourceShareInvitationArn, nil
+			}
+		}
+
+		if out.NextToken == nil {
+			break
+		}
+		nextToken = aws.StringValue(out.NextToken)
+	}
+
+	return nil, fmt.Errorf("Unable to find a pending invitation for resource share %s", resourceShareARN)
+}
+
+func resourceAwsRamResourceShareAccepterRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ramconn
+
+	request := &ram.GetResourceShareInvitationsInput{
+		ResourceShareInvitationArns: []*string{aws.String(d.Id())},
+	}
+
+	out, err := conn.GetResourceShareInvitations(request)
+	if err != nil {
+		if isAWSErr(err, ram.ErrCodeUnknownResourceException, "") {
+			log.Printf("[WARN] No RAM resource share invitation by ARN (%s) found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading RAM resource share invitation %s: %s", d.Id(), err)
+	}
+
+	if len(out.ResourceShareInvitations) == 0 {
+		log.Printf("[WARN] No RAM resource share invitation by ARN (%s) found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	invitation := out.ResourceShareInvitations[0]
+
+	d.Set("status", invitation.Status)
+	d.Set("receiver_account_id", invitation.ReceiverAccountId)
+	d.Set("sender_account_id", invitation.SenderAccountId)
+	d.Set("share_arn", invitation.ResourceShareArn)
+	d.Set("invitation_arn", invitation.ResourceShareInvitationArn)
+	d.Set("share_id", resourceAwsRamResourceShareGetIDFromARN(*invitation.ResourceShareArn))
```

Comment:
Use aws.StringValue() instead of dereferencing directly.

Reply:
Actually, I don't see `share_id` in any of the other RAM resources and having to compute the ID from the ARN seems a bit janky, can we just remove this attribute?